### ISSUE TITLE
Feature request: no validation datasets, sharing validation set

### ISSUE THREAD

**Issue by sherifshehata** [NVIDIA/DIGITS#1102]
Hello,
I would suggest that you redesign the handling of training and validation sets. In some cases a user may need to do training only without validation (maybe because the user merged validation and training sets for generating a final model, or to do faster training). Additionally sometimes the user could use a shared validation set across different versions of training sets, which would lead to data duplication in the current setup.

Thanks you.

**Comment by gheinrich** [comment]
Hi @sherifshehata are you referring to the Object Detection workflow?

**Comment by sherifshehata** [comment]
Yes. Sorry i didn't specify that in the original description.

**Comment by gheinrich** [comment]
The original intention was to make the validation set optional in the object detection work-flow. Obviously now if you leave the corresponding fields blank in the form you get an error. Digging into this a bit more it looks like [validate_required_if_set](https://github.com/NVIDIA/DIGITS/blob/v4.0.0/digits/utils/forms.py#L51) is not behaving as intended because of this test:

```python
if other_field_value is not None and other_field_value is not "":
```

In our case if we leave the field blank the value of the field is `u''` and:
```python
>>> u'' is not ''
True
```

I don't know if it's a regression (I think I tested it!) but it looks to me like we could replace the condition with:
```python
if other_field_value:
```

As it seems to do what we want:
```python
>>> bool(None)
False
>>> bool(u'')
False
>>> bool('')
False
```

I have tentatively pushed #1105 if you wish to try. @lukeyeager any idea on this?

**Comment by sherifshehata** [comment]
I think this will solve it. Thank you.
